File: shared/types/src/sync.ts
// Sync types for delta push/pull

export interface SyncItem {
  id: string;
  encrypted_data: string;
  folder_id?: string | null;
  item_type?: number;
  version: number;
  updated_at: string;
  deleted_at?: string | null;
}

export interface SyncFolder {
  id: string;
  encrypted_name: string;
  parent_id?: string | null;
  updated_at: string;
  deleted_at?: string | null;
}

export interface SyncPushRequest {
  device_id: string;
  items: SyncItem[];
  folders: SyncFolder[];
}

export interface SyncConflict {
  id: string;
  server_version: number;
  server_updated_at: string;
}

export interface SyncPushResponse {
  accepted_items: string[];
  accepted_folders: string[];
  conflicts: SyncConflict[];
}

export interface SyncPullResponse {
  items: SyncItem[];
  folders: SyncFolder[];
  deleted_ids: string[];
  server_time: string;
  has_more: boolean;
  next_cursor?: string;
}

export interface SyncQueueEntry {
  id: string;
  item_id: string;
  action: 'create' | 'update' | 'delete';
  timestamp: number;
}

export type SyncStatus = 'idle' | 'syncing' | 'error' | 'disabled';

File: _archive/crates/vaultic-types/src/sync.rs
//! Sync-related types for delta push/pull.

use chrono::{DateTime, Utc};
use serde::{Deserialize, Serialize};
use uuid::Uuid;

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct SyncPushRequest {
    pub device_id: String,
    pub items: Vec<SyncItem>,
    pub folders: Vec<SyncFolder>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct SyncItem {
    pub id: Uuid,
    pub folder_id: Option<Uuid>,
    pub item_type: Option<i16>,
    pub encrypted_data: String,
    pub version: i32,
    pub updated_at: DateTime<Utc>,
    pub deleted_at: Option<DateTime<Utc>>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct SyncFolder {
    pub id: Uuid,
    pub encrypted_name: String,
    pub parent_id: Option<Uuid>,
    pub updated_at: DateTime<Utc>,
    pub deleted_at: Option<DateTime<Utc>>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct SyncPushResponse {
    pub accepted_items: Vec<Uuid>,
    pub accepted_folders: Vec<Uuid>,
    pub conflicts: Vec<SyncConflict>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct SyncConflict {
    pub id: Uuid,
    pub server_version: i32,
    pub server_updated_at: DateTime<Utc>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct SyncPullResponse {
    pub items: Vec<SyncItem>,
    pub folders: Vec<SyncFolder>,
    pub deleted_ids: Vec<Uuid>,
    pub server_time: DateTime<Utc>,
}
